code/modules/mob/living/carbon/carbon.dm

```diff
@@ -725,16 +725,18 @@
 		add_logs(src, C, "devoured")
 
 /mob/living/carbon/proc/create_bodyparts()
+	var/l_arm_index_next = -1
+	var/r_arm_index_next = 0
 	for(var/X in bodyparts)
 		var/obj/item/bodypart/O = new X()
 		O.owner = src
 		bodyparts.Remove(X)
 		bodyparts.Add(O)
 		if(O.body_part == ARM_LEFT)
-			O.held_index = 1
+			O.held_index = (l_arm_index_next += 2) //1, 3, 5, 7...

```

Comment:
Yes it does compile, I tested it locally before I even suggested the idea.
511 upwards, which we should be targeting by now (heck, i think it's already on the servers, @MrStonedOne ?)